# Case Study 01: Sales Force Training

### Perusahaan X ingin meningkatkan penjualan mereka. Dari data penjualan sebelumnya menunjukkan bahwa penjualan rata-rata yaitu $100 per transaksi. Setelah melakukan training kepada pekerja sales, didapatkan data penjualan terbaru. Projek ini bertujuan menganalisa penjualan rata-rata, apakah naik atau tidak.

1.   Buatlah DataFrame 1 kolom dengan 25 baris bernama ‘TransactionAmount’
2.   Buatlah analisis measure of central tendency dari data ‘TransactionAmount’
3. Lakukan juga analisis measure of variability dari data ‘TransactionAmount’
4. Tentukan H0 & H1 untuk persiapan dilakukan T-Test
5. Lakukan proses T-Test pada data rata-rata jumlah transaksi (dengan alpha = 5%)
6. Buatlah kesimpulan dari hasil pengujian sales force training!
7. Membuat Power point sebagai dokumen pengumpulan tugas dari proyek yang telah dikerjakan, serta mengupload code dan power point projeknya pada github repository







## __Import Libraries__

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats

## _1) Membuat array sebagai dataset penjualan rata-rata sebanyak 25 baris, dengan nama kolom TransactionAmount_

In [ ]:
ta = [100, 150, 50, 100, 130,
      120, 100, 85, 70, 150,
      150, 120, 50, 100, 100,
      140, 90, 150, 50, 90,
      120, 100, 110, 75, 65]
DataFrame1 = pd.DataFrame(ta, columns=['Transaction_Amount']) #Memasukan list ta ke dalam dataframe DataFrame1 dengan nama kolom TransactionAmount,
DataFrame1


Transaction_Amount
0                  100
1                  150
2                   50
3                  100
4                  130
5                  120
6                  100
7                   85
8                   70
9                  150
10                 150
11                 120
12                  50
13                 100
14                 100
15                 140
16                  90
17                 150
18                  50
19                  90
20                 120
21                 100
22                 110
23                  75
24                  65

## 2) Analisis measure of central tendency dari data ‘TransactionAmount’

Ringkasan yang mendeskripsikan keseluruhan kumpulan data, dengan satu nilai yang mewakili titik tengah atau pusat distribusinya.

In [ ]:
Mean = DataFrame1['Transaction_Amount'].mean() #menampilkan nilai mean
Median = DataFrame1['Transaction_Amount'].median() #menampilkan nilai median/tengah
Mode = DataFrame1['Transaction_Amount'].mode()[0]#menampilkan nilai modus/terbanyak
print("Mean:", Mean)
print("Median:", Median)
print("Mode:", Mode)

Mean: 102.6
Median: 100.0
Mode: 100


## 3) Analisis measure of variability dari data ‘TransactionAmount’

Menggambarkan penyebaran atau dispersi dari sekumpulan data.

In [ ]:
variance = df['TransactionAmount'].var()
std_dev = df['TransactionAmount'].std()
print("Variance:", variance)
print("Standard Deviation:", std_dev)
mx = DataFrame1['Transaction_Amount'].max()
mn = DataFrame1['Transaction_Amount'].min()
print("Maximum:", mx)
print("Minimum:", mn)
print("Range:", mx - mn)
Q1 = DataFrame1['Transaction_Amount'].quantile(0.25)
Q2 = DataFrame1['Transaction_Amount'].quantile(0.50)
Q3 = DataFrame1['Transaction_Amount'].quantile(0.75)


print("Q1:", Q1)
print("Q2:", Q2)
print("Q3:", Q3)



Variance: 1012.7499999999997
Standard Deviation: 31.82373328193912
Maximum: 150
Minimum: 50
Range: 100
Q1: 85.0
Q2: 100.0
Q3: 120.0


## 4) Tentukan H0 & H1 untuk persiapan dilakukan T-Test
Hipotesis untuk T-Test

H0: Rata-rata jumlah transaksi sama dengan 100
H1: Rata-rata jumlah transaksi lebih besar dari dengan 100
## 5) Lakukan proses T-Test pada data rata-rata jumlah transaksi (dengan alpha = 5%)


# Mendefinisikan nilai alpha sebesar 5%

In [ ]:
#stats.ttest_1samp(DataFrame1['Transaction_Amount'], 100)
#stats.ttest_1samp(a= DataFrame1, popmean=100)

TtestResult(statistic=array([0.40850016]), pvalue=array([0.68652848]), df=array([24]))

In [ ]:
t_stat, p_value = stats.ttest_1samp(DataFrame1['Transaction_Amount'], 100)
alpha = 0.05
reject = p_value < alpha
print("T-Statistic:", t_stat)
print("P-Value:", p_value)



T-Statistic: 0.4085001556802841
P-Value: 0.6865284813438117


## 6) Kesimpulan

In [ ]:
kesimpulan = "Tolak H0" if reject else "Gagal menolak H0"
print(kesimpulan)

Gagal menolak H0


- P-value lebih besar daripada alpha (5%), jadi gagal menolak H0
- Training tidak berhasil, rata-rata penjualan masih $100

In [ ]:
#H1 lebih besar
#stats.t.ppf(1-0.05, 24) #24 greed of fredom n-1 (25-1)
cr = cr = stats.t.ppf(1-0.05, 24)
print("Critical Value:", cr)


Critical Value: 1.7108820799094275


Critical Value lebih besar dibanding T-Statistic dengan jumlah 0.4085001556802841. Maka tetap gagal untuk menolak H0, dimana training tidak berpengaruh pada peningkatan penjualan.